controlplane/src/core/bufservices/PlatformService.ts

```diff
@@ -5412,22 +5412,43 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
       return handleError<PlainMessage<GetLatestValidSubgraphSDLByNameResponse>>(logger, async () => {
         const authContext = await opts.authenticator.authenticate(ctx.requestHeader);
         const subgraphRepo = new SubgraphRepository(opts.db, authContext.organizationId);
-        const federatedGraphRepo = new FederatedGraphRepository(opts.db, authContext.organizationId);
+
+        req.namespace = req.namespace || DefaultNamespace;
 
         const subgraph = await subgraphRepo.byName(req.name, req.namespace);
-        const federatedGraph = await federatedGraphRepo.byName(req.fedGraphName, req.namespace);
-        if (!subgraph || !federatedGraph) {
+        if (!subgraph) {
```

Comment:
We have 2 rpc methods to achieve this. `getLatestSubgraphSDLByNameRequest` and `GetLatestValidSubgraphSDLByName`. I suggest to call the methods conditionally on CLI and revert changes to this one.

Reply:
Good catch! We need to become better in naming :smile:

Reply:
Done. I also improved naming. We haven't used the methods anywhere except in the Studio.